### ISSUE TITLE
feature request: JSON Schema binding/generation/validation

### ISSUE THREAD

**Issue by autonome** [cocopon/tweakpane#531]
This maybe should be a plug-in, but I am filing anyway in case others follow down this path...

I'm using JSON Schema for data validation, and also Pane generation.

For example, I wrote a function which generates a new Pane based on a JSON Schema definition.

The generated Pane is a form for creating a new instance of the schema object. Then I validate the resulting data (pulled manually until #431 is fixed) against the schema.

If I pass data along with the schema, the Pane will be filled with data for editing by the user.

This way I can rapidly build fairly robust interfaces with Tweakpane which conform to external data definitions.

Having schema support built-in somehow may be worth considering.

**Comment by cocopon** [comment]
Right. It's not primitive and it should be handled by the plugin system as you said.

BTW I use issues mainly to handle bugs. I'd like to keep the issue list clean but I can't close feature requests. I think [GitHub Discussions](https://github.com/cocopon/tweakpane/discussions) is the better place to discuss features.
